applications/asset_tracker/src/cloud_codec/cloud_codec.c

```diff
@@ -0,0 +1,385 @@
+		.children = _children,	\
+		.num_children = ARRAY_SIZE(_children),	\
+	}
+
+#define CMD_NEW_GROUP(_name, _group, _children)		\
+	struct cmd _name = {		\
+		.name = STRINGIFY(_name),		\
+		.group = _group,			\
+		.children = _children,			\
+		.num_children = ARRAY_SIZE(_children),	\
+	};
+
+
+static CMD_NEW_GROUP(set, CLOUD_CMD_GROUP_SET, CMD_ARRAY(
+	CMD_NEW_RECIPIENT(environment, CLOUD_RCPT_ENVIRONMENT, CMD_ARRAY(
+		CMD_NEW_CHAN(humidity, CLOUD_CHANNEL_HUMID, CMD_ARRAY(
+			CMD_NEW_TYPE(enable, CLOUD_CMD_ENABLE),
+			CMD_NEW_TYPE(disable, CLOUD_CMD_DISABLE),
+			CMD_NEW_TYPE(threshold_high, CLOUD_CMD_THRESHOLD_HIGH),
+			CMD_NEW_TYPE(threshold_low, CLOUD_CMD_THRESHOLD_LOW))
+		),
+		CMD_NEW_CHAN(pressure, CLOUD_CHANNEL_AIR_PRESS, CMD_ARRAY(
+			CMD_NEW_TYPE(enable, CLOUD_CMD_ENABLE),
+			CMD_NEW_TYPE(disable, CLOUD_CMD_DISABLE),
+			CMD_NEW_TYPE(threshold_high, CLOUD_CMD_THRESHOLD_HIGH),
+			CMD_NEW_TYPE(threshold_low, CLOUD_CMD_THRESHOLD_LOW))
+		),
+		CMD_NEW_CHAN(temperature, CLOUD_CHANNEL_TEMP, CMD_ARRAY(
+			CMD_NEW_TYPE(enable, CLOUD_CMD_ENABLE),
+			CMD_NEW_TYPE(disable, CLOUD_CMD_DISABLE),
+			CMD_NEW_TYPE(threshold_high, CLOUD_CMD_THRESHOLD_HIGH),
+			CMD_NEW_TYPE(threshold_low, CLOUD_CMD_THRESHOLD_LOW))
+		),
+		CMD_NEW_CHAN(air_quality, CLOUD_CHANNEL_AIR_QUAL, CMD_ARRAY(
+			CMD_NEW_TYPE(enable, CLOUD_CMD_ENABLE),
+			CMD_NEW_TYPE(disable, CLOUD_CMD_DISABLE),
+			CMD_NEW_TYPE(threshold_high, CLOUD_CMD_THRESHOLD_HIGH),
+			CMD_NEW_TYPE(threshold_low, CLOUD_CMD_THRESHOLD_LOW))
+		))
+	),
+	CMD_NEW_RECIPIENT(motion, CLOUD_RCPT_MOTION, CMD_ARRAY(
+		CMD_NEW_CHAN(flip, CLOUD_CHANNEL_FLIP, CMD_ARRAY(
+			CMD_NEW_TYPE(enable, CLOUD_CMD_ENABLE),
+			CMD_NEW_TYPE(disable, CLOUD_CMD_DISABLE),
+			CMD_NEW_TYPE(threshold_high, CLOUD_CMD_THRESHOLD_HIGH),
+			CMD_NEW_TYPE(threshold_low, CLOUD_CMD_THRESHOLD_LOW))
+		),
+		CMD_NEW_CHAN(impact, CLOUD_CHANNEL_IMPACT, CMD_ARRAY(
+			CMD_NEW_TYPE(enable, CLOUD_CMD_ENABLE),
+			CMD_NEW_TYPE(disable, CLOUD_CMD_DISABLE),
+			CMD_NEW_TYPE(threshold_high, CLOUD_CMD_THRESHOLD_HIGH),
+			CMD_NEW_TYPE(threshold_low, CLOUD_CMD_THRESHOLD_LOW))
+		))
+	),
+	CMD_NEW_RECIPIENT(ui, CLOUD_RCPT_UI, CMD_ARRAY(
+		CMD_NEW_CHAN(pin, CLOUD_CHANNEL_PIN, CMD_ARRAY(
+			CMD_NEW_TYPE(enable, CLOUD_CMD_ENABLE),
+			CMD_NEW_TYPE(disable, CLOUD_CMD_DISABLE),
+			CMD_NEW_TYPE(pwm, CLOUD_CMD_PWM))
+		),
+		CMD_NEW_CHAN(led, CLOUD_CHANNEL_RGB_LED, CMD_ARRAY(
+			CMD_NEW_TYPE(red, CLOUD_CMD_LED_RED),
+			CMD_NEW_TYPE(green, CLOUD_CMD_LED_GREEN),
+			CMD_NEW_TYPE(blue, CLOUD_CMD_LED_BLUE),
+			CMD_NEW_TYPE(pulse_length, CLOUD_CMD_LED_PULSE_LENGTH),
+			CMD_NEW_TYPE(pause, CLOUD_CMD_LED_PAUSE_LENGTH))
+		),
+		CMD_NEW_CHAN(buzzer, CLOUD_CHANNEL_BUZZER, CMD_ARRAY(
+			CMD_NEW_TYPE(enable, CLOUD_CMD_ENABLE),
+			CMD_NEW_TYPE(disable, CLOUD_CMD_DISABLE),
+			CMD_NEW_TYPE(play_melody, CLOUD_CMD_PLAY_MELODY),
+			CMD_NEW_TYPE(play_note, CLOUD_CMD_PLAY_NOTE))
+		))
+	))
+);
+
+static CMD_NEW_GROUP(get, CLOUD_CMD_GROUP_GET, CMD_ARRAY(
```

Comment:
Let's prefix the group variables with `group` to make them more readable when used elsewhere in the code

Reply:
OK